pkg/sql/schemachanger/scbuild/internal/scbuildstmt/alter_table_alter_column_set_default.go

```diff
@@ -0,0 +1,42 @@
+// Copyright 2024 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package scbuildstmt
+
+import (
+	"github.com/cockroachdb/cockroach/pkg/sql/schemachanger/scpb"
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
+)
+
+func alterTableSetDefault(
+	b BuildCtx, tn *tree.TableName, tbl *scpb.Table, t *tree.AlterTableSetDefault,
+) {
+	alterColumnPreChecks(b, tn, tbl, t.Column)
+	colID := getColumnIDFromColumnName(b, tbl.TableID, t.Column, true /* required */)
+	// Block alters on system columns.
+	scpb.ForEachColumn(
+		b.QueryByID(tbl.TableID),
+		func(_ scpb.Status, _ scpb.TargetStatus, e *scpb.Column) {
+			if e.ColumnID == colID {
+				// Block drops on system columns.
+				panicIfSystemColumn(e, t.Column.String())
+			}
+		})
```

Comment:
nit: I think scpb.ForEachColumn is deprecated. Let's use something like 

```
panicIfSystemColumn(mustRetrieveColumnElem(b, tbl.TableID, colID), t.Column.String())
```

and then kindly update `mustRetrieveColumnElem` to use the now-preferred way of retrieving a column element
```
func mustRetrieveColumnElem(
	b BuildCtx, tableID catid.DescID, columnID catid.ColumnID,
) (column *scpb.Column) {
	return b.QueryByID(tableID).FilterColumn().Filter(func(
		current scpb.Status, target scpb.TargetStatus, e *scpb.Column,
	) bool {
		return e.ColumnID == columnID
	}).MustGetOneElement()
}
```

Reply:
done

Reply:
I changed this in `ALTER ... SET NOT NULL` as well (first commit)